src/Microsoft.Extensions.Configuration.AzureAppConfiguration/AzureAppConfigurationProvider.cs

```diff
@@ -29,7 +29,8 @@ internal class AzureAppConfigurationProvider : ConfigurationProvider, IConfigura
         private readonly bool _requestTracingEnabled;
         private readonly IConfigurationClientManager _configClientManager;
         private AzureAppConfigurationOptions _options;
-        private Dictionary<string, ConfigurationSetting> _applicationSettings;
+        private Dictionary<string, ConfigurationSetting> _serverData;

```

Comment:
I was wondering if we had to store `ConfigurationSetting` in there or if it could be a type that just has key, label and etag. Unfortunately, it looks like the SDK requires `ConfigurationSetting` as a param. I'd like to see if we could work with SDK team to get an overload that doesn't require us to keep the entire `ConfigurationSetting` object around.

Reply:
I was wondering if we had to store `ConfigurationSetting` in there or if it could be a type that just has key, label and etag. Unfortunately, it looks like the SDK requires `ConfigurationSetting` as a param. I'd like to see if we could work with SDK team to get an overload that doesn't require us to keep the entire `ConfigurationSetting` object around.

Reply:
Looks like we can with the [preview version of the SDK](https://learn.microsoft.com/en-us/dotnet/api/azure.data.appconfiguration.configurationclient.getconfigurationsettingasync?view=azure-dotnet-preview#azure-data-appconfiguration-configurationclient-getconfigurationsettingasync(system-string-system-string-system-string-system-collections-generic-ienumerable((system-string))-azure-matchconditions-azure-requestcontext))

Reply:
Also, it doesn't look like we need to store every key-value in _serverData. It's only used for refreshing key-value collections.

Going further, if it's only being used for change detection, why isn't it merged with _watchedSettings?

Reply:
Going off what we talked about last week, I noticed that trying to pass in null for the value parameter for the settings that I deep copied into watchedsettings ended up failing some unit tests.

Reply:
But we didn't aim to nullify values in this PR right?